src/lib/Strategy/Storage/GroupItemStrategyDispatcher.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzRecommendationClient\Strategy\Storage;
+
+use EzSystems\EzRecommendationClient\Exception\UnsupportedGroupItemStrategy;
+use Ibexa\Contracts\Personalization\Criteria\CriteriaInterface;
+use Ibexa\Contracts\Personalization\Storage\DataSourceInterface;
+use Ibexa\Contracts\Personalization\Value\ItemGroupListInterface;
+use Traversable;
+
+final class GroupItemStrategyDispatcher implements GroupItemStrategyDispatcherInterface
+{
+    /** @var iterable<\EzSystems\EzRecommendationClient\Strategy\Storage\GroupItemStrategyInterface> */
+    private iterable $strategies;
+
+    /** @param iterable<\EzSystems\EzRecommendationClient\Strategy\Storage\GroupItemStrategyInterface> $strategies */
+    public function __construct(iterable $strategies)
+    {
+        $this->strategies = $strategies;
+    }
+
+    public function getGroupList(
+        DataSourceInterface $source,
+        CriteriaInterface $criteria,
+        string $groupBy
+    ): ItemGroupListInterface {
+        $strategies = $this->strategies instanceof Traversable
+            ? iterator_to_array($this->strategies)
+            : $this->strategies;
+
+        if (!array_key_exists($groupBy, $strategies)) {
```

Comment:
tbh I find the `isset` variant more readable (maybe I'm more used to it).

```suggestion
        if (!isset($strategies[$groupBy])) {
```

The difference is minor (`isset` does not allow null value, and there is no longer a performance penalty for `array_key_exists` since PHP7, I think).